src/hotspot/share/classfile/classLoaderExt.cpp

```diff
@@ -213,6 +213,12 @@ void ClassLoaderExt::process_jar_manifest(JavaThread* current, ClassPathEntry* e
     char sep = os::file_separator()[0];
     const char* dir_name = entry->name();
     const char* dir_tail = strrchr(dir_name, sep);
+#ifdef _WINDOWS
+    // On Windows, we also support forward slash as the file separator for Class-Path: attribute.
+    if (dir_tail == nullptr) {
+      dir_tail = strrchr(dir_name, '/');
```

Comment:
There's still a bug in the current patch. If the input is

```
java -cp a\b/c.jar
```

The current patch gets "a" as the directory name, but it should get "a\b" instead.

Reply:
I've pushed an updated fix to address this above. Also added a test.